File: Java Solution/ProjectEuler2.java
public class ProjectEuler2{
    public static void main(String[] args){
        
        int sum = 0;

        int pre = 1, cur = 1, next = 2;

        while(cur < 4000000){
            if (cur % 2 == 0){
                sum += cur;
            }
            pre = cur;
            cur = next;
            next = pre + cur;

        }

        System.out.println(sum);

    }
}

File: CPP Solution/ProjectEuler_2.cpp
#include <iostream>
using namespace std;

int main(){

    long long int previous = 1, current = 1, next = 2, sum = 0;

    while (current < 4000000){
        if (current % 2 == 0){
            sum += current;
        }
        previous = current;
        current = next;
        next = previous + current;
    }

    cout << sum << endl;

    return 0;
}